src/Serilog.Enrichers.CorrelationId/Middlewares/CorrelationIdHeaderSupplierMiddleware.cs

```diff
@@ -0,0 +1,44 @@
+#if NETCORE
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Http;
+
+namespace Serilog.Enrichers.CorrelationId.Middlewares
+{
+    public class CorrelationIdHeaderSupplierMiddleware
+    {
+        private readonly RequestDelegate _next;
+        private readonly string _headerKey;
+
+        public CorrelationIdHeaderSupplierMiddleware(RequestDelegate next, string headerKey = "x-correlation-id")
+        {
+            _next = next;
+            _headerKey = headerKey;
+        }
+
+        public Task Invoke(HttpContext httpContext)
+        {
+            string correlationId = Guid.NewGuid().ToString();
+            if (httpContext.Request.Headers.TryGetValue(_headerKey, out var values))
+            {
+                correlationId = values.First();
```

Comment:
as per above comment, this could then be `correlationId = values.First() ?? Guid.NewGuid().ToString();`

Reply:
fixed in d667c8e